lib/spack/spack/util/module_cmd.py

```diff
@@ -24,6 +23,26 @@
 def module(*args, **kwargs):
     module_cmd = kwargs.get("module_template", "module " + " ".join(args))
 
+    # Use the actual environment module function call as there is no guarantee that the bash
+    # function `module` is available.
+    if not any(x.startswith("BASH_FUNC_module") for x in os.environ):
+        if os.environ.get("MODULES_CMD", None):
+            module_cmd = kwargs.get(
+                "module_template", 'eval `${MODULES_CMD} sh "' + " ".join(args) + '"`'
+            )
+
+        elif os.environ.get("LMOD_CMD", None):
+            module_cmd = kwargs.get(
+                "module_template",
+                'eval "$($LMOD_CMD shell "'
+                + " ".join(args)
+                + '")" && eval "$(${LMOD_SETTARG_CMD:-:} -s sh)"',
```

Comment:
If I read the docs correctly, you just need `LMOD_CMD` and check that its value isn't `:` (that's how they "disable" the command if not run interactively, apparently)

Reply:
OK.